packages/transaction-controller/src/TransactionController.ts

```diff
@@ -556,14 +621,117 @@ export class TransactionController extends BaseControllerV1<
 
     this.addPendingTransactionTrackerListeners();
 
+    // when transactionsController state changes
+    // check for pending transactions and start polling if there are any
+    this.subscribe(this.#checkForPendingTransactionAndStartPolling);
+
+    // TODO once v2 is merged make sure this only runs when
+    // selectedNetworkClientId changes
     onNetworkStateChange(() => {
       log('Detected network change', this.getChainId());
+      this.pendingTransactionTracker.onStateChange();
       this.onBootCleanup();
     });
 
     this.onBootCleanup();
+    this.messagingSystem.subscribe(
+      'NetworkController:stateChange',
+      (_, patches) => {
+        if (this.enableMultichain) {
+          const networkClients = this.getNetworkClientRegistry();
+          patches.forEach(({ op, path }) => {
+            if (op === 'remove' && path[0] === 'networkConfigurations') {
+              const networkClientId = path[1] as NetworkClientId;
+              delete networkClients[networkClientId];
+            }
+          });
+
+          this.#refreshTrackingMap(networkClients);
+          this.#refreshEtherscanRemoteTransactionSources(networkClients);
+        }
+      },
+    );
+    if (this.enableMultichain) {
+      this.#initTrackingMap();
+    }
   }
 
+  setEnableMultichain(enableMultichain: boolean) {
+    this.enableMultichain = enableMultichain;
+    if (enableMultichain) {
+      this.#initTrackingMap();
+    }
+  }
+
+  /**
+   * Stops polling and removes listeners to prepare the controller for garbage collection.
+   */
+  destroy() {
```

Comment:
Is this required in our controller lifecycle?

I've always assumed all controllers are designed to be persist for the lifetime of the clients.

Reply:
🤔 This appears to be a standard pattern for our controllers and one that is enshrined in [the BaseControllerV2](https://github.com/MetaMask/core/blob/7b497fa71ecd785174dc6e818a945f1e415e8f00/packages/base-controller/src/BaseControllerV2.ts#L242) which @mcmire has [a PR to introduce to this controller](https://github.com/MetaMask/core/pull/3827) after this PR is merged.